flink-autoscaler/src/main/java/org/apache/flink/autoscaler/ScalingExecutor.java

```diff
@@ -229,13 +229,14 @@ Map<JobVertexID, ScalingSummary> computeScalingSummary(
                                 var currentParallelism =
                                         (int) metrics.get(ScalingMetric.PARALLELISM).getCurrent();
 
-                                final boolean hasKeyBy =
-                                        jobTopology.get(v).getInputs().containsValue(HASH);
+                                var inputs = jobTopology.get(v).getInputs();
+                                var adjustByMaxParallelism =
+                                        inputs.isEmpty() || inputs.containsValue(HASH);
                                 var newParallelism =
                                         jobVertexScaler.computeScaleTargetParallelism(
                                                 context,
                                                 v,
-                                                hasKeyBy,
+                                                adjustByMaxParallelism,
                                                 metrics,
```

Comment:
Can we please add a single unit test case to cover this core piece of the logic? Just to make sure we don't mess up sources in the future unintentionally

Reply:
Good suggestion, added~

Reply:
Thank you very much for your review, especially on the weekend. ❤️